src/librepgp/stream-parse.cpp

```diff
@@ -758,8 +781,49 @@ signed_validate_signature(pgp_source_signed_param_t *param, pgp_signature_info_t
 static void
 signed_src_update(pgp_source_t *src, const void *buf, size_t len)
 {
+    if (!len) {
+        return;
+    }
     pgp_source_signed_param_t *param = (pgp_source_signed_param_t *) src->param;
     pgp_hash_list_update(param->hashes, buf, len);
+    /* update text-mode sig hashes */
+    if (param->txt_hashes.empty()) {
+        return;
+    }
+    /* check whether we had CR at the end of last chunk and LF at the beginning */
+    uint8_t *ch = (uint8_t *) buf;
+    if (param->lastcr && (*ch == CH_LF)) {
+        ch++;
+    }
+    uint8_t *linebeg = ch;
+    uint8_t *end = (uint8_t *) buf + len;
```

Comment:
@rrrooommmaaa Thanks for pointing at this. While it should not happen in real-life systems, this may be hit on low-memory embedded systems or so on. Added a commit.

Reply:
So how would `signed_src_update(src, &last, 1)` work when the last char is at the end of addressable space?
I suggest simply to replace `ch < end` and `ch + 1 < end` with `!=`
I know that `<` is generally safer than `!=` but in this case it would be neater

Reply:
@rrrooommmaaa That was the first idea, however then we would end up in need to check whether ch++ overflows, adding one more check on each byte processed.

Reply:
Ok, merging then.